RELEASE.md

```diff
@@ -1,3 +1,21 @@
+The 2.14 minor series tracks TensorFlow 2.14.
+
+## Features
+
+- Small data table and step selector usability and accessibility improvements (#6409, #6410, #6431)
```

Comment:
Can you call out #6372? You could add it to the list here or consider mentioning it in its own line. I think it's the most noteworthy of improvements to the data table.

Reply:
Done.